src/effects/VST3/VST3Effect.cpp

```diff
@@ -329,64 +329,36 @@ bool VST3Effect::SupportsAutomation() const
 }
 
 bool VST3Effect::SaveSettings(
-   const EffectSettings &, CommandParameters & parms) const
+   const EffectSettings& settings, CommandParameters& parms) const
 {
-   if(mEditController == nullptr)
-      return false;
+   const auto& vst3Settings = GetSettings(settings);
 
-   using namespace Steinberg;
-   
-   for(int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
-   {
-      Vst::ParameterInfo parameterInfo { };
-      if(mEditController->getParameterInfo(i, parameterInfo) == kResultOk)
-      {
-         if(parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
-         {
-            parms.Write(
-               VST3Utils::MakeAutomationParameterKey(parameterInfo),
-               mEditController->getParamNormalized(parameterInfo.id)
-            );
-         }
-      }
-   }
+   if (vst3Settings.mProcessorStateStr)
+      parms.Write( processorStateKey,  wxString( vst3Settings.mProcessorStateStr.value() )  );
+
+   if (vst3Settings.mControllerStateStr)
+      parms.Write( controllerStateKey, wxString( vst3Settings.mControllerStateStr.value() ) );
 
    return true;
 }
 
 bool VST3Effect::LoadSettings(
-   const CommandParameters & parms, EffectSettings &settings) const
+   const CommandParameters& parms, EffectSettings& settings) const
```

Comment:
`LoadSettings`, like `FetchSettings`, should make the given settings reflect exactly what is in the instance or the config file.  So, add other branches that assign nullopt when the configuration does not contain a string.

Reply:
ok, done